ArtOfIllusion/src/artofillusion/LayoutWindow.java

```diff
@@ -45,30 +45,41 @@ Changes copyright (C) 2017-2020 by Petri Ihalainen
   SceneViewer theView[];
   BorderContainer viewPanel[];
   FormContainer viewsContainer;
-  FormContainer centerContainer;
+  
   private DockingContainer dock[];
   BScrollPane itemTreeScroller;
   Score theScore;
   ToolPalette tools;
   private TexturesAndMaterialsDialog assetsDialog;
-  BLabel helpText;
+  private final BLabel helpText = new BLabel();
   TreeList itemTree;
   Scene theScene;
-  BMenuBar menubar;
-  BMenu fileMenu, recentFilesMenu, editMenu, objectMenu, createMenu, toolsMenu, viewMenu, scriptMenu;
-  BMenu animationMenu, editKeyframeMenu, sceneMenu;
+
+  private final BMenu fileMenu = Translate.menu("file");
+  private final BMenu editMenu = Translate.menu("edit");
+  private final BMenu sceneMenu = Translate.menu("scene");
+  private final BMenu objectMenu = Translate.menu("object");
+  private final BMenu animationMenu = Translate.menu("animation");
+  private final BMenu toolsMenu = Translate.menu("tools");
+  private final BMenu viewMenu = Translate.menu("view");
+  
+  
+  private BMenu recentFilesMenu, scriptMenu;
+  
```

Comment:
We also need to be careful of changing access here. I know of at least one plugin that directly accesses the menus and rearranges things.

Reply:
Which plugin may affect this?

Reply:
ViewLayout for certain. The point is, this is a breaking change. What is the migration plan?

Reply:
Take a brief look at ViewLayout code...
The plugin does not touches this Layout Window fields at all
at first it gets MenuBar via BFrame getMenuBar() than accesses Menu and Menu item via indexes.

Reply:
Thinking about it more, your response just makes me even more skeptical that this change is good or beneficial. Since any plugin *could* get around it by calling the `getMenuBar()`, why pretend that these fields are *not* accessible to any code that can get an instance of the layout window?

Reply:
All LayoutWindow root menu items are accessible via getters.

Reply:
How is that an improvement over a public field?

Reply:
Accessing fields via getters is better solution from incapsulation POV. Anoter reason is that fields now is not public but has only package private access. So hide them is a good decisoin for me